src/medCore/database/medDatabaseRemover.cpp

```diff
@@ -311,7 +320,12 @@ void medDatabaseRemover::removePatient ( int patientDbId )
     if ( query.next() )
     {
         QString thumbnail = query.value ( 0 ).toString();
-        this->removeFile ( thumbnail );
+
+        medAbstractDbController * dbc = medDataManager::instance()->controllerForDataSource(d->index.dataSourceId());
+        if (thumbnail == dbc->metaData(d->index,  medMetaDataKeys::ThumbnailPath.key()))
+        {
+            this->removeFile ( thumbnail );
+        }
```

Comment:
Three times the exact same code. This could have been put in a separate function.